docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+| migration processed memory         | hypervisor    | yes       |             |
+| migration remaining memory         | hypervisor    | yes       |             |
+| migration memory bps               | hypervisor    | yes       |             |
+| migration memory constant [TODO]   | hypervisor    | yes       |             |
+| migration memory compression bytes | hypervisor    | yes       |             |
+| migration dirty rate               | hypervisor    | yes       |             |
+| migration memory iteration         | hypervisor    | yes       |             |
+| guest total memory                 | guest agent   | -         |             |
+| guest free memory                  | guest agent   | -         |             |
+| guest CPU count                    | guest agent   | -         |             |
+| guest free disk space              | guest agent   | -         |             |
+| guest IP address                   | guest agent   | -         |             |
+
+
+**NOTE** the list of guest agent provided metrics is partial, because
+a lot of different metrics can be collected for the guest. The presented
+entries are a subset of what the [QEMU Guest Agent](https://wiki.qemu.org/Features/GuestAgent)
+provides, which is used by virtualization managers like
+[oVirt](http://www.ovirt.org)
+
+From now on, the overlapping metrics  (aka 'system metrics') will be ignored,
+and we will focus on the 'hypervisor' and 'guest agent' metrics.
+
+## collection of VM metrics
+A natural fit for the kubevirt specific metrics is the [custom metrics](
+https://github.com/kubernetes/kubernetes/blob/release-1.2/docs/proposals/custom-metrics.md).
+All the metrics are accessible through the libvirt API.
+
+The collection of the VM metrics will be done by a separate pod, implemented as kubernetes DaemonSet.
+This is because the metrics are available from a centralized source, the libvirt daemon.
+To increase resilience, each node should expose all the metrics it knows about, avoiding
+a single point of failure per cluster. The resulting architecture is shown in the figure below.
+
+![VM metrics diagram](vm-metrics.png "VM Metrics")
+
+To have the full set the VM metrics, a guest agent like the QEMU Guest Agent should be run
+in the virtual machines. This step, albeit recommended, is totally left up to the administrator
+of the cluster.
+
+
+## VM metrics naming rules
+
+In order to minimize the clashes and make the VM metrics name predictable, we adopt the following rules
+to set the names:
+
+1. A name should be in the form `{VM name}.{group}.{metric name}`. The dot (`.`) is used as separator.
+   the `camelCase` should be used for composite words. Example: `VM0.migration.totalData`
+
+2. Abbreviations should be used only if their meaning is well known and not ambiguous. Examples:
+
+- **BAD**:
+  `VM0.migration.bps`
+  What `bps` is? Is it Bytes Per Second or Bits per Second? Or something else more specific?
```

Comment:
Yes, eventually encoding the unit in the name might make sense.

Reply:
bps is definitely bits per seconds btw :) Byte uses uppercase B when the unit is written properly. But it should not have its own leaf, unless you want to expose multiple different scales.